lib/configurators/aws/aws.go

```diff
@@ -61,50 +61,55 @@ const (
 	defaultAttachUser = "username"
 )
 
+type databaseActions struct {
+	// discovery is a list of actions used for database discovery.
+	discovery []string
+	// iamAuth is a list of actions used for enableing IAM auth.
+	iamAuth []string
+	// metadata is a list of actions used for fetching database metadata
+	// (excluding the ones already in "discovery").
+	metadata []string
+	// managedUsers is a list of actions used for managing database users.
+	managedUsers []string
+	// boundary is a list of actions only used for boundary policies.
+	boundary []string
+
+	requireIAMEdit        bool
+	requireSecretsManager bool
+}
+
+func (a databaseActions) buildStatement(discoveryService, boundary bool) *awslib.Statement {
```

Comment:
I would prefer value receivers here.

Reply:
Understood, I thought it was by mistake. Why do you prefer a value receiver here? Just curious

Reply:
The main reason for me is `databaseActions` is not meant to be modified so it's clearer using a values receiver. It's a relatively small struct (copying a small []string shouldn't be a big deal). I recall values receivers use stack instead of heap so it makes things slightly easier for garbage collection.